mmocr/datasets/pipelines/loading.py

```diff
@@ -66,3 +67,39 @@ def _load_masks(self, results):
         results['gt_masks'] = gt_masks
         results['mask_fields'].append('gt_masks')
         return results
+
+
+@PIPELINES.register_module()
+class LoadImageFromMat(LoadImageFromFile):
+    """Load an image from cv Mat.
+
+    Similar with :obj:`LoadImageFromFile`, but the image read from
+    ``results['img']``, which is np.ndarray.
+    """
+
+    def __call__(self, results):
+        """Call functions to add image meta information.
+
+        Args:
+            results (dict): Result dict with Webcam read image in
+                ``results['img']``.
+
+        Returns:
+            dict: The dict contains loaded image and meta information.
+        """
+
+        img = results['img']
+        if self.to_float32:
```

Comment:
mv this after L97, since float32 will affect how bgr2gray or gray2bgr works.

For simplicity, you can assert that results['img'] is uint8, since otherwise, things might go wrong with color conversions